src/librustc_typeck/diagnostics.rs

```diff
@@ -1864,12 +1864,35 @@ fn main<T>() { // error: main function is not allowed to have type parameters
 "##,
 
 E0132: r##"
+A function with `start` attribute was declared with type parameters.
+
+Erroneous code example:
+
+```compile_fail
+#![feature(start)]
+
+#[start]
+fn f<T>() {}
+```
+
 It is not possible to declare type parameters on a function that has the `start`
-attribute. Such a function must have the following type signature:
+attribute. Such a function must have the following type signature (which is the
+same than C main functions):
```

Comment:
I would say "the same as C `main` functions", but also, we generally try not to make comparisons to other languages, generally, so I'm sympathetic but also hesitant